osquery/tables/system/linux/deb_packages.cpp

```diff
@@ -7,196 +7,100 @@
  * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
  */
 
-// see README.api of libdpkg-dev
-#define LIBDPKG_VOLATILE_API
-
-#include <boost/algorithm/string.hpp>
-
 #include <osquery/core/system.h>
 #include <osquery/core/tables.h>
-#include <osquery/filesystem/filesystem.h>
 #include <osquery/logger/logger.h>
+#include <osquery/utils/linux/idpkgquery.h>
 #include <osquery/worker/ipc/platform_table_container_ipc.h>
 #include <osquery/worker/logging/glog/glog_logger.h>
 
-extern "C" {
-#include <dpkg/dpkg-db.h>
-#include <dpkg/dpkg.h>
-#include <dpkg/parsedump.h>
-#include <dpkg/pkg-array.h>
-}
-
 namespace osquery {
 namespace tables {
 
-static const std::string kDPKGPath{"/var/lib/dpkg"};
+namespace {
 
-/// A comparator used to sort the packages array.
-int pkg_sorter(const void* a, const void* b) {
-  const struct pkginfo* pa = *(const struct pkginfo**)a;
-  const struct pkginfo* pb = *(const struct pkginfo**)b;
-  const char* arch_a = pa->installed.arch->name;
-  const char* arch_b = pb->installed.arch->name;
+// From the documentation: this directory contains many files that give
+// information about status of installed or uninstalled packages
+const std::string kAdminDir{"/var/lib/dpkg"};
 
-  int res = strcmp(pa->set->name, pb->set->name);
-  if (res != 0) {
-    return res;
-  }
+void logError(Logger& logger,
+              const std::string& message,
+              const Error<IDpkgQuery::ErrorCode>& error,
+              const std::string& admindir) {
+  auto error_code = error.getErrorCode();
 
-  if (pa->installed.arch == pb->installed.arch) {
-    return 0;
-  }
+  const auto& error_description =
+      IDpkgQuery::getErrorCodeDescription(error_code);
 
-  return strcmp(arch_a, arch_b);
+  logger.vlog(1,
```

Comment:
If this is an actual error, do you think it belongs on verbose or info?

Reply:
> If this is an actual error, do you think it belongs on verbose or info?

I think it probably makes sense to turn this into a real error. Thanks for spotting this!